drivers/pds/lib/utils.go

```diff
@@ -2427,3 +2436,194 @@ func UpdateDeploymentResourceConfig(deployment *pds.ModelsDeployment, namespace
+func ExpandAndValidatePxPool(context []*scheduler.Context) error {
+	log.InfoD("Entering into PX-POOL Expansion...")
+	log.FailOnError(err, "Failed to list storage pools")
+	poolIDToResize := pickPoolToResize(context)
+	poolToBeResized := getStoragePool(poolIDToResize)
+	log.InfoD("Pool to be resized is %v", poolToBeResized)
+
+	log.InfoD("Verify that pool resize is not in progress")
+	if val, err := poolResizeIsInProgress(poolToBeResized); val {
+		// wait until resize is completed and get the updated pool again
+		poolToBeResized, err = GetStoragePoolByUUID(poolIDToResize)
+		log.FailOnError(err, fmt.Sprintf("Failed to get pool using UUID %s", poolIDToResize))
+	} else {
+		log.FailOnError(err, fmt.Sprintf("pool [%s] cannot be expanded due to error: %v", poolIDToResize, err))
+	}
+	var expectedSize uint64
+	var expectedSizeWithJournal uint64
+	log.InfoD("Calculate expected pool size and trigger pool resize")
+	expectedSize = poolToBeResized.TotalSize * 2 / units.GiB
+	expectedSize = roundUpValue(expectedSize)
+	isjournal, err := IsJournalEnabled()
+	log.FailOnError(err, "Failed to check is Journal enabled")
+
+	expectedSizeWithJournal = expectedSize
+	if isjournal {
+		expectedSizeWithJournal = expectedSizeWithJournal - 3
+	}
+
+	log.InfoD("Current Size of the pool %s is %d", poolIDToResize, poolToBeResized.TotalSize/units.GiB)
+
+	err = tests.Inst().V.ExpandPool(poolIDToResize, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, false)
+
+	resizeErr := waitForPoolToBeResized(expectedSize, poolIDToResize, isjournal)
+	if resizeErr != nil {
+		log.FailOnError(resizeErr, fmt.Sprintf("Failed to resize pool with ID-  %s", poolIDToResize))
+	}
+	ValidateApplications(context)
+	resizedPool, err := GetStoragePoolByUUID(poolIDToResize)
+	log.FailOnError(err, fmt.Sprintf("Failed to get pool using UUID %s", poolIDToResize))
+	newPoolSize := resizedPool.TotalSize / units.GiB
+	isExpansionSuccess := false
+	if newPoolSize >= expectedSizeWithJournal {
+		isExpansionSuccess = true
+	} else {
+		if ResiliencyFlag {
+			CapturedErrors <- err
+		}
+		log.FailOnError(err, fmt.Sprintf("Failed to resize pool with ID-  %s", poolIDToResize))
+		return err
+	}
+	log.InfoD("Px-Pool expansion status is - %v", isExpansionSuccess)
+	log.InfoD("expected new pool size to be %v or %v if pool has journal, got %v", expectedSize, expectedSizeWithJournal, newPoolSize)
+	return nil
+}
+
+func getStoragePool(poolIDToResize string) *api.StoragePool {
+	pool, err := GetStoragePoolByUUID(poolIDToResize)
+	log.FailOnError(err, "Failed to get pool using UUID %s", poolIDToResize)
+	return pool
+}
+
+func pickPoolToResize(contexts []*scheduler.Context) string {
+	poolWithIO, err := GetPoolIDWithIOs(contexts)
+	if poolWithIO == "" || err != nil {
+		log.Warnf("No pool with IO found, picking a random pool in use to resize")
+	}
+	poolIDsInUseByTestingApp, err := GetPoolsInUse()
+	log.FailOnError(err, "Error identifying pool to run test")
+	poolIDToResize := poolIDsInUseByTestingApp[0]
```

Comment:
Do you want to always pick only one pool ID for this test case ? Can we send this entire list and resize all of those pools - that will be another good test case. Upto you how you want to handle this.